apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/ErrorReporting/ErrorReportingContextExceptionLogger.cs

```diff
@@ -83,64 +91,69 @@ void IContextExceptionLogger.Log(Exception exception, IContextWrapper context)
 
         /// <summary>
         /// Gets information about the HTTP request and response when the exception occurred 
-        /// and populates a <see cref="HttpRequestContext"/> object.
+        /// and populates a <see cref="Struct"/>.
         /// </summary>
-        private HttpRequestContext CreateHttpRequestContext(IContextWrapper context)
+        private Struct CreateHttpRequestContext(IContextWrapper context)
         {
-            return new HttpRequestContext()
-            {
-                Method = context?.GetHttpMethod() ?? "",
-                Url = context?.GetUri() ?? "",
-                UserAgent = context?.GetUserAgent() ?? "",
-                ResponseStatusCode = context?.GetStatusCode() ?? 0,
-            };
+            Struct httpRequestContext = new Struct();
+            httpRequestContext.Fields.Add("method", Value.ForString(context?.GetHttpMethod() ?? ""));
+            httpRequestContext.Fields.Add("url", Value.ForString(context?.GetUri() ?? ""));
+            httpRequestContext.Fields.Add("userAgent", Value.ForString(context?.GetUserAgent() ?? ""));
+            httpRequestContext.Fields.Add("responseStatusCode", Value.ForNumber(context?.GetStatusCode() ?? 0));
+            return httpRequestContext;
         }
 
         /// <summary>
         /// Gets information about the source location where the exception occurred 
-        /// and populates a <see cref="SourceLocation"/> object.
+        /// and populates a <see cref="Struct"/>.
         /// </summary>
-        private static SourceLocation CreateSourceLocation(Exception exception)
+        private static Struct CreateSourceLocation(Exception exception)
         {
             if (exception == null)
             {
-                return new SourceLocation();
+                return new Struct();
             }
 
             StackTrace stackTrace = new StackTrace(exception, true);
             StackFrame[] frames = stackTrace.GetFrames();
             if (frames == null || frames.Length == 0)
             {
-                return new SourceLocation();
+                return new Struct();
             }
 
             StackFrame frame = frames[0];
-            return new SourceLocation
-            {
-                FilePath = frame.GetFileName() ?? "",
-                LineNumber = frame.GetFileLineNumber(),
-                FunctionName = frame.GetMethod()?.Name ?? "",
-            };
+            Struct sourceLocation = new Struct();
```

Comment:
This can be written as

```csharp
Struct sourceLocation = new Struct {
    Fields = {
        { "filePath", Value.ForString(frame.GetFileName() ?? "") },
        { "lineNumber", Value.ForNumber(frame.GetFileLineNumber()) },
        { "functionName", Value.ForString(frame.GetMethod()?.Name ?? "" }
    }
}
```

(Or you could use indexers instead.)